src/app/routes/article/article.service.ts

```diff
@@ -270,7 +285,9 @@ export const getArticle = async (slug: string, id?: number) => {
     throw new HttpException(404, { errors: { article: ['not found'] } });
   }
 
-  return articleMapper(article, id);
+  const mappedArticle = articleMapper(article, id);
+  await cache.set(redis, cacheKey, mappedArticle);
```

Comment:
```
const mappedArticle = articleMapper(article, id);
await cache.set(redis, cacheKey, mappedArticle);
```

These cache set lines seem to repeat over and over again - they should be extracted to a func or even all the way to the caching implementation eg. `cache.set(slug, mappedArticle)` or even `cache.set(article)` which then also leads to a the leaky abstraction of the cache implementation itself (this service should not know that it is a specific caching implementation eg. `redis` in this case.